a3p-integration/proposals/p:upgrade-19/provisionPool.test.js

```diff
@@ -0,0 +1,230 @@
+ *   6c. Try to open a vault WITHOUT provisioning the newly introduced account
+ *   6d. Observe the new account's address under `published.wallet`
+ * 7. Same as step 2. Checks manual provision works after null upgrade
+ */
+
+import '@endo/init';
+import test from 'ava';
+import { execFileSync } from 'node:child_process';
+import {
+  addUser,
+  makeAgd,
+  evalBundles,
+  agd as agdAmbient,
+  agoric,
+  getISTBalance,
+  getDetailsMatchingVats,
+  VALIDATORADDR,
+  CHAINID,
+  GOV1ADDR,
+  openVault,
+  ATOM_DENOM,
+} from '@agoric/synthetic-chain';
+import {
+  makeVstorageKit,
+  retryUntilCondition,
+  waitUntilAccountFunded,
+  waitUntilContractDeployed,
+} from '@agoric/client-utils';
+import { NonNullish } from '@agoric/internal';
+
+const PROVISIONING_POOL_ADDR = 'agoric1megzytg65cyrgzs6fvzxgrcqvwwl7ugpt62346';
+
+const ADD_PSM_DIR = 'addUsdLemons';
+const DEPOSIT_USD_LEMONS_DIR = 'depositUSD-LEMONS';
+const UPGRADE_PP_DIR = 'upgradeProvisionPool';
+const NULL_UPGRADE_PP_DIR = 'nullUpgradePP';
+
+const USDC_DENOM = NonNullish(process.env.USDC_DENOM);
+
+const agd = makeAgd({ execFileSync }).withOpts({ keyringBackend: 'test' });
+
+const ambientAuthority = {
+  query: agdAmbient.query,
+  follow: agoric.follow,
+  setTimeout,
+  log: console.log,
+};
+
+/**
+ * @param {string} addr
+ * @param {string} wanted
+ * @param {string} [from]
+ */
+export const bankSend = (addr, wanted, from = VALIDATORADDR) => {
+  return agd.tx(['bank', 'send', from, addr, wanted], {
+    chainId: CHAINID,
+    from,
+    yes: true,
+  });
+};
+
+const provision = (name, address) =>
+  agd.tx(['swingset', 'provision-one', name, address, 'SMART_WALLET'], {
+    chainId: 'agoriclocal',
+    from: 'validator',
+    yes: true,
+  });
+
+const introduceAndProvision = async name => {
+  const address = await addUser(name);
+  console.log('ADDR', name, address);
+
+  const provisionP = provision(name, address);
+
+  return { provisionP, address };
+};
+
+/**
+ * @param {import('@agoric/client-utils').VstorageKit} vstorageKit
+ */
+const getProvisionedAddresses = async vstorageKit => {
+  const children = await vstorageKit.vstorage.keys('published.wallet');
+  return children;
+};
+
+const checkUserProvisioned = (addr, vstorageKit) =>
+  retryUntilCondition(
+    () => getProvisionedAddresses(vstorageKit),
+    children => children.includes(addr),
+    'Account not provisioned',
+    { maxRetries: 5, retryIntervalMs: 1000, log: console.log, setTimeout },
+  );
+
+test.before(async t => {
+  const vstorageKit = await makeVstorageKit(
+    { fetch },
+    { rpcAddrs: ['http://localhost:26657'], chainName: 'agoriclocal' },
+  );
+
+  t.context = {
+    vstorageKit,
+  };
+});
+
+test.serial('upgrade provisionPool', async t => {
+  await evalBundles(UPGRADE_PP_DIR);
+
+  const vatDetailsAfter = await getDetailsMatchingVats('provisionPool');
+  const { incarnation } = vatDetailsAfter.find(vat => vat.vatID === 'v28'); // provisionPool is v28
```

Comment:
In a3p and mainNet, this is v28, but that might not always be true. `getDetailsMatchingVats()` returns `name` as the exact name of the vat, so you can validate by comparing the name using `===`.

 I didn't fix `getVatDetails`, so that one returns the `name` provided as an argument, and wouldn't distinguish "provisionPool" from "provisionPool-governor". 

Oh, wait. `getVatDetails` might return names like "zcf-b1-db93f-provisionPool" and "zcf-b1-9f877-provisionPool-governor", so you might have to verify with `name.endsWith('provisionPool')`.

Reply:
Alright, I'll switch to validating by vat-name instead of vatID 👍